### PR TITLE
WIP: GCP: Add a metadata startup-script to update a VM's dns resolver

### PR DESCRIPTION
**Pull request by sadasu**
There might be a way to use the terraform module https://github.com/terraform-google-modules/terraform-google-startup-scripts.  Choosing this approach now for its simplicity.

### CODE REVIEW

File: data/data/gcp/cluster/master/main.tf [hunk 1]
```diff
@@ -85,6 +85,7 @@ resource "google_compute_instance" "master" {
   metadata = {
     user-data = var.ignition
   }
+  metadata_startup_script = "echo \"Custom-DNS Test\"; echo \"Added by OpenShift\" >> /tmp/temp.conf; echo \"nameserver $(ip route get 8.8.8.8 | grep -oP 'src \\K[^ ]+')\" >> /tmp/temp.conf; cat /etc/resolv.conf >> /tmp/temp.conf; cat /tmp/temp.conf; sudo cp /tmp/temp.conf /etc/resolv.conf"
```
Comment (barbacbd) [inline data/data/gcp/cluster/master/main.tf:88]: Do we use any startup scripts on control-plane nodes? If so this is going to overwrite it. 

Also, do we need the custom dns test here? This is being applied no matter what configuration is chosen, so the print statements may not be appropriate.

Comment (sadasu) [reply data/data/gcp/cluster/master/main.tf]: > Do we use any startup scripts on control-plane nodes? If so this is going to overwrite it.

Agreed. If we do, I was expecting it to be set in this file. I didn't see any so assuming we don't. Is there any other way to find out?

Added the conditional now.

**Comment by sadasu** [comment]
From testing, it appears that the metadata's startup-script is being ignored when Ignition is provided via user-data. Further, the "Note" at the bootom of https://docs.fedoraproject.org/en-US/fedora-coreos/provisioning-gcp/ confirms what I am seeing while testing this.

**Comment by sadasu** [comment]
Not using this approach

**Pull request closed without merging**